istioctl/cmd/proxyconfig.go

```diff
@@ -993,3 +1014,25 @@ func getPodNameBySelector(labelSelector string) ([]string, string, error) {
 	ns = pl.Items[0].Namespace
 	return podNames, ns, nil
 }
+
+func getPodsNameInDefaultNamespace(toComplete string) ([]string, error) {
+	kubeClient, err := kubeClient(kubeconfig, configContext)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx := context.Background()
+	podList, err := kubeClient.CoreV1().Pods(defaultNamespace).List(ctx, metav1.ListOptions{})
```

Comment:
The `getPodName` is [already available](https://github.com/istio/istio/blob/a9d6b12c44557cc1893a54a202dbcabe7a5e7c0a/istioctl/cmd/proxyconfig.go#L965-L978) and you can directly use it here. 

The `proxy-config` can also look into other namespaces and not just the default namespace. That func already handles custom namespace logic.

Reply:
yeah.. This feature is a candy for users to auto completion pods name in current namespace, that's because the command use `<pod-name[.namespace]>` format to specify a pod, that is difficult for completion as the pod-name is before namespace name.
`istioctl proxy-config clusters <pod-name[.namespace]>`

So it is a little heavy to use the `getPodName` func as we only want to get pods in current namespace.

Reply:
Ok, can you try installing Istio in a custom namespace other than `istio-system` and then run `proxy-config` to try auto-completion for pods?

Reply:
Thanks for your advice, I will reinstall istio in other namespace in my env later.
However I am not sure what this is to verify for, as the `defaultNamespace` is current-context's namespace specified in kubeconfig, maybe `default`, `istio-demo`, etc,  it has no relationship with `istio-system`.

Reply:
@shamsher31 
I installed istio in namespace `istio-x`, and proxy-config auto-completion still works fine.